xesmf/frontend.py

```diff
@@ -1183,6 +1184,21 @@ def __init__(
             self._lon_out_name = 'lon'
             self._lat_out_name = 'lat'
 
+        # Check length of polys segments, issue warning if too long
+        check_polys, check_holes, _, _ = split_polygons_and_holes(polys)
+        check_polys.extend(check_holes)
+        poly_segments = []
+        for check_poly in check_polys:
+            b = check_poly.boundary.coords
+            poly_segments.extend(
+                [LineString(b[k : k + 2]).length for k in range(len(b) - 1)]
+            )  # Length of each segment
+        if np.any(np.array(poly_segments) > 1.0):
+            warnings.warn(
+                'Polys contain large (>1deg) segments. This could lead to unsuspected errors. To prevent errors over large region, please densify your polys using the densify_poly function',
+                UserWarning,
+            )
+
```

Comment:
I suggest this goes into a stand-alone function.

Reply:
Good idea, I'll make the change